selene_sdk/samplers/dataloader.py

```diff
@@ -0,0 +1,308 @@
+"""
+This module provides the `SamplerDataLoader` and  `SamplerDataSet` classes,
+which allow parallel sampling for any Sampler or FileSampler using
+torch DataLoader mechanism.
+"""
+import  sys
+
+import h5py
+import numpy as np
+import torch
+from torch.utils.data import Dataset, DataLoader
+
+
+class _SamplerDataset(Dataset):
+    """
+    This class provides a Dataset interface that wraps around a Sampler or
+    FileSampler. `_SamplerDataset` is used internally by `SamplerDataLoader`.
+
+    Parameters
+    ----------
+    sampler : selene_sdk.samplers.Sampler or
+        selene_sdk.samplers.file_samplers.FileSampler
+        The sampler from which to draw data.
+
+    Attributes
+    ----------
+    sampler : selene_sdk.samplers.Sampler or
+            selene_sdk.samplers.file_samplers.FileSampler
+        The sampler from which to draw data.
+    """
+    def __init__(self, sampler):
+        super(_SamplerDataset, self).__init__()
+        self.sampler = sampler
+
+    def __getitem__(self, index):
+        """
+        Retrieve sample(s) from self.sampler. Only index length affects the
+        number of samples. The index values are not used.
+
+        Parameters
+        ----------
+        index : int or any object with __len__ method implemented
+            The size of index is used to determine the number of the
+            samples to return.
+
+        Returns
+        ----------
+        sequences, targets : tuple(numpy.ndarray, numpy.ndarray)
+            A tuple containing the numeric representation of the
+            sequence examples and their corresponding labels. The
+            shape of `sequences` will be
+            :math:`I \\times L \\times N`, where :math:`I` is
+            `index`, :math:`L` is the sequence length, and
+            :math:`N` is the size of the sequence type's alphabet.
+            The shape of `targets` will be :math:`I \\times T`,
+            where :math:`T` is the number of targets predicted.
+        """
+        sequences, targets = self.sampler.sample(
+            batch_size=1 if isinstance(index, int) else len(index))
+        if sequences.shape[0] == 1:
+            sequences = sequences[0,:]
+            targets = targets[0,:]
+        return sequences, targets
+
+    def __len__(self):
+        """
+        Implementing __len__ is required by the DataLoader. So as a workaround,
+        this returns `sys.maxsize` which is a large integer which should
+        generally prevent the DataLoader from reaching its size limit.
+
+        Another workaround that is implemented is catching the StopIteration
+        error while calling `next` and reinitialize the DataLoader.
+        """
+        return sys.maxsize
+
+
+class SamplerDataLoader(DataLoader):
+    """
+    A DataLoader that provides parallel sampling for any `Sampler`
+    or `FileSampler` object. SamplerDataLoader requires sampler to be
+    initialized with `pickleable=True` to enable multi-procesing.
```

Comment:
is this a parameter in samplers that I couldn't find (`picklable=True`)? do you initialize like this by default now?

Reply:
yes this is default and not configurable now - I will change the doc